### DISCUSSION TITLE
How to prevent redirect to /api/auth/verify-request?provider=resend&type=email after Magic Link?

### DISCUSSION THREAD

**Discussion by subproject22** [nextauthjs/next-auth#11101] [Help]
After I sent email with `resend` provider, it redirects me to `/api/auth/verify-request?provider=resend&type=email`. 

Why it's doing it? I don't wan't to go anywhere. I want to show user success message on same page, or redirect to any other page I want. How can I do that? I've tried to do that in `signIn` action and then in `sendVerificationRequest` but it's not working or throwing redirect error (tried `redirect`)

**Comment by KhaledGharib** [comment]
Hey @subproject22
It seems that you are trying to avoid this default behavior: redirecting to `/api/auth/verify-request?provider=resend&type=email` after sending a Magic Link. You would like to display a success message or redirect to another page.

You will be able to achieve this by using the `redirect` property in the `signIn` method. The default value is `true`. It turns off default behavior for redirection if set to `false`. Here's an example:
```typescript
"use server";
import { signIn } from "@/auth";

export const handleEmailSignIn = async (email: string) => {
  try {
    await signIn("nodemailer", {
      email,
      callbackUrl: "/dashboard", // URL to redirect after successful sign-in
      redirect: false, // Prevent default redirect behavior
    });
  } catch (error) {
    throw error;
  }
};
``` 
```typescript
  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    try {
      await handleEmailSignIn(email); // Server Action
      toast("Check your email for the Magic Link");
    } catch (error) {
      console.error(error);
    }
  };

``` 

If you want to redirect users to custom pages, you can specify these pages in your ./auth.ts configuration file like so:
```typescript
  pages: {
    signIn: "/auth/sign-in",
    error: "/auth/auth-error",
  },
```
For more details, you can check the documentation here: [Custom Pages](https://authjs.dev/getting-started/session-management/custom-pages)

Hope this helps 🤗

**Reply by Julian24Design** [reply]
When `redirect` is set to `false`, `signIn` doesn't throw any errors—they're just logged on the server. Is there a way to handle this?

**Comment by rafal-r** [comment]
There used the be possibility to customise this route with `verifyRequest` prop under pages options in v4.
https://next-auth.js.org/configuration/pages

But I'm not sure if it still available in v5.
